models/work-item-link-repository.go

```diff
@@ -0,0 +1,180 @@
+		fmt.Printf("\n\nError: %s\n\n", db.Error.Error())
+		return nil, NewInternalError(fmt.Sprintf("Failed to find work item link type: %s", db.Error.Error()))
+	}
+
+	// Fetch the source work item
+	source := WorkItem{}
+	db = r.db.Where("id=?", link.SourceID).Find(&source)
+	if db.RecordNotFound() {
+		return nil, NewBadParameterError("source work item", strconv.FormatUint(link.SourceID, 10))
+	} else if db.Error != nil {
+		return nil, NewInternalError(fmt.Sprintf("Failed to find source work item: %s", db.Error.Error()))
+	}
+
+	// Fetch the target work item
+	target := WorkItem{}
+	db = r.db.Where("id=?", link.TargetID).Find(&target)
+	if db.RecordNotFound() {
+		return nil, NewBadParameterError("target work item", strconv.FormatUint(link.TargetID, 10))
+	} else if db.Error != nil {
+		return nil, NewInternalError(fmt.Sprintf("Failed to find target work item: %s", db.Error.Error()))
+	}
+
+	// Check that the work item types of the source and target match those specified in the link type
+	if source.Type != linkType.SourceTypeName {
+		return nil, NewBadParameterError("source.type", source.Type).Expected(linkType.SourceTypeName)
+	}
+	if target.Type != linkType.TargetTypeName {
+		return nil, NewBadParameterError("target.type", target.Type).Expected(linkType.TargetTypeName)
+	}
+
+	db = db.Create(link)
+	if db.Error != nil {
+		return nil, NewInternalError(db.Error.Error())
+	}
+
+	// Convert the created link type entry into a JSONAPI response
+	result := ConvertLinkFromModel(link)
+	return &result, nil
+}
+
+// Load returns the work item link for the given ID.
+// Returns NotFoundError, ConversionError or InternalError
+func (r *GormWorkItemLinkRepository) Load(ctx context.Context, ID string) (*app.WorkItemLink, error) {
+	id, err := satoriuuid.FromString(ID)
+	if err != nil {
+		// treat as not found: clients don't know it must be a UUID
+		return nil, NewNotFoundError("work item link", ID)
+	}
+	log.Printf("loading work item link %s", id.String())
+	res := WorkItemLink{}
+	//db := r.db.Model(&res).Where("id=?", ID).First(&res)
+	db := r.db.Where("id=?", id).Find(&res)
+	if db.RecordNotFound() {
+		log.Printf("not found work item link, res=%v", res)
+		return nil, NewNotFoundError("work item link", id.String())
+	}
+	// Convert the created link type entry into a JSONAPI response
+	result := ConvertLinkFromModel(&res)
+	return &result, nil
+}
+
+// List returns all work item links
+// TODO: Handle pagination
+func (r *GormWorkItemLinkRepository) List(ctx context.Context) (*app.WorkItemLinkArray, error) {
+	// We don't have any where clause or paging at the moment.
+	var rows []WorkItemLink
+	db := r.db.Find(&rows)
+	if db.Error != nil {
+		return nil, db.Error
+	}
+	res := app.WorkItemLinkArray{}
+	res.Data = make([]*app.WorkItemLinkData, len(rows))
+	for index, value := range rows {
+		cat := ConvertLinkFromModel(&value)
+		res.Data[index] = cat.Data
+	}
+	// TODO: When adding pagination, this must not be len(rows) but
```

Comment:
We really should add paging & querying here ASAP, as link count is going to get big fast.